signer/src/keys.rs

```diff
@@ -262,6 +264,20 @@ impl fake::Dummy<fake::Faker> for PublicKey {
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
 pub struct PrivateKey(secp256k1::SecretKey);
 
+impl FromStr for PrivateKey {
+    type Err = Error;
+
+    /// Attempts to parse a [`PrivateKey`] from the hex representation of a
+    /// a secp256k1 private key.
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let bytes = hex::decode(s).unwrap();
+        if bytes.len() != 32 {
+            return Err(Error::InvalidPrivateKeyLength(bytes.len()));
+        }
+        PrivateKey::from_slice(&hex::decode(s).unwrap()[0..32])
```

Comment:
I think this could be simplified to 
```rust
secp256k1::SecretKey::from_str(s)
    .map(PrivateKey)
    .map_err(Error::InvalidPrivateKey)
```

Reply:
On second thought, what you have here is better since it tells the user what the issue was. `rust-secp256k1` only tells you that the key was invalid, and there are two or three reasons for that. But we should not `unwrap` here. Instead we should wrap the error variant and then `?`. Alternatively we could either skip this impl entirely since we do not need it, or we go with `secp256k1::SecretKey::from_str(s)`. I'm fine with which ever.

Reply:
Yeah I missed to go back and clean this one up -- now it's better :)

While purely functionally it's not needed, it _is_ needed for decoding from a `&str` and getting more informative errors, as you mention, which is _always_ better when the output is user-facing.